### ISSUE TITLE
[BLOCKER] Implement Automatic Feature Selection with Importance Scoring

### ISSUE THREAD

**Issue by frankbria** [frankbria/narrative-modeling-app#73]
## 🎯 Overview
Implement automated feature selection algorithms that identify and rank the most important features for the predictive model.

## 📊 Current State
- No way to identify which features are most predictive
- Users may include too many or too few features
- No guidance on feature importance

## 🎯 Expected Outcome
The system should:
- Automatically identify most important features
- Rank features by predictive power
- Remove redundant or low-value features
- Explain why features are important or not
- Support multiple selection algorithms
- Allow user override of automatic selections

## ✅ Acceptance Criteria
- [ ] Implement multiple feature selection algorithms:
  - Correlation-based selection
  - Mutual information
  - Random Forest importance
  - Recursive Feature Elimination (RFE)
  - LASSO regularization
  - Statistical tests (chi-square, ANOVA)
- [ ] Feature importance dashboard showing:
  - Ranked list of features
  - Importance scores (0-1 scale)
  - Visual importance bars
  - Correlation heatmap
  - Feature redundancy analysis
- [ ] Selection controls:
  - Auto-select top N features
  - Set importance threshold
  - Manual include/exclude overrides
  - Compare different selection methods
- [ ] For each feature, show:
  - Importance score
  - Selection status (included/excluded)
  - Reason for inclusion/exclusion
  - Correlation with other features
  - Correlation with target variable
- [ ] Export selected feature set for model training

### Frontend (TypeScript/Next.js)
- Components:
  - \`FeatureSelection\`: Main selection interface
  - \`FeatureImportanceChart\`: Visual ranking of features
  - \`FeatureCorrelationHeatmap\`: Correlation matrix visualization
  - \`SelectionControls\`: Algorithm and threshold controls
  - \`FeatureComparison\`: Compare selection methods side-by-side
  - \`SelectedFeatureSet\`: Final selection summary
- Interactive visualizations (D3.js or Chart.js)
- State management for feature selection

### Backend (Python/FastAPI)
- Service:
  ```python
  class FeatureSelectionService:
      async def select_features(
          self,
          dataset_id: str,
          target_column: str,
          method: str,
          params: Dict[str, Any]
      ) -> FeatureSelectionResult:
          \"\"\"Select features using specified method\"\"\"
          pass
      
      async def calculate_importance(
          self,
          X: pd.DataFrame,
          y: pd.Series,
          method: str
      ) -> Dict[str, float]:
          \"\"\"Calculate feature importance scores\"\"\"
          pass
      
      async def detect_redundancy(
          self,
          X: pd.DataFrame,
          threshold: float = 0.95
      ) -> List[Tuple[str, str, float]]:
          \"\"\"Detect highly correlated feature pairs\"\"\"
          pass
      
      async def explain_selection(
          self,
          feature: str,
          importance: float,
          correlations: Dict[str, float]
      ) -> str:
          \"\"\"Generate explanation for feature selection\"\"\"
          pass
  
  class FeatureSelectionResult(BaseModel):
      selected_features: List[str]
      feature_scores: Dict[str, float]
      method: str
      explanation: str
      redundant_pairs: List[Tuple[str, str, float]]
      correlation_matrix: List[List[float]]
  ```
- Feature selection algorithms:
  - Correlation-based: scipy.stats.pearsonr
  - Mutual information: sklearn.feature_selection.mutual_info_classif
  - Tree-based: sklearn.ensemble.RandomForestClassifier.feature_importances_
  - RFE: sklearn.feature_selection.RFE
  - LASSO: sklearn.linear_model.LassoCV
  - Statistical: sklearn.feature_selection.chi2, f_classif
- Endpoints:
  - \`POST /api/datasets/{id}/features/select\` - Run feature selection
  - \`POST /api/datasets/{id}/features/importance\` - Calculate importance
  - \`POST /api/datasets/{id}/features/redundancy\` - Detect redundancy
  - \`GET /api/datasets/{id}/features/selected\` - Get selected features

### Performance Optimization
- Use sampling for large datasets in importance calculation
- Cache correlation matrices
- Parallel computation for multiple algorithms
- Progressive loading for large feature sets

## 📋 Subtasks
- [ ] Implement correlation-based selection
- [ ] Implement mutual information selection
- [ ] Implement tree-based importance
- [ ] Implement RFE algorithm
- [ ] Implement LASSO selection
- [ ] Implement statistical tests
- [ ] Create importance scoring interface
- [ ] Build correlation heatmap visualization
- [ ] Implement redundancy detection
- [ ] Create feature selection dashboard
- [ ] Add algorithm comparison interface
- [ ] Implement explanation generator
- [ ] Create unit tests for each algorithm
- [ ] Create integration tests for selection workflow
- [ ] Add performance tests for large feature sets
- [ ] Update documentation

## 🔗 Dependencies
- #71 (AI-driven feature suggestions) - provides features to select from
- #72 (Visual feature builder) - provides custom features to select from

## 📚 Related Issues
- AI-driven feature suggestions (#71)
- Visual feature builder (#72)
- Feature store (to be created)

## 🏷️ Labels
\`blocker\`, \`backend\`, \`frontend\`, \`ml-algorithms\`, \`stage-4-feature-engineering\`

## ⏱️ Estimated Effort
3-4 weeks

## 📝 Notes
- Critical for model performance optimization
- Multiple algorithms provide different perspectives
- Explain feature selection in business terms
- Allow manual override for domain expertise
- Performance optimization essential for large feature sets
- Follow TDD approach with >85% test coverage
